test/dotcom/system_status/groups_test.exs

```diff
@@ -0,0 +1,632 @@
+defmodule Dotcom.SystemStatus.GroupsTest do
+  use ExUnit.Case, async: true
+  doctest Dotcom.SystemStatus.Groups
+
+  alias Dotcom.SystemStatus.Groups
+  alias Test.Support.Factories.Alerts.Alert
+  alias Test.Support.Factories.Alerts.InformedEntity
+  alias Test.Support.Factories.Alerts.InformedEntitySet
+
+  @all_rail_lines ["Blue", "Green", "Orange", "Red"]
+  @green_line_branches ["Green-B", "Green-C", "Green-D", "Green-E"]
+  @heavy_rail_lines ["Blue", "Orange", "Red"]
```

Comment:
We can export these out of the module.

```
defmodule Dotcom.SystemStatus.Groups do
  @heavy_rail_lines ~w[Blue Orange Red]
  @all_rail_lines @heavy_rails_lines ++ ~w[Green]
  
  def heavy_rail_lines, do: @heavy_rail_lines
  
  def all_rail_lines, do: @all_rail_lines
end

defmodule Dotcom.SystemStatus.GroupsTest do
  describe "groups/2" do
    test "sorts groups by heavy rail first" do
      # Exercise
      groups = Group.groups([], time_today())
      
      # Verify
      route_ids = groups |> Enum.map(&1.route_id)
      heavy_rail_lines = heavy_rail_lines()
      assert Enum.take(route_ids, Kernel.length(heavy_rail_lines)) == heavy_rail_lines
    end
  end
end
```

Reply:
I see that that's possible, but what is the benefit?

My problem is that it makes the tests a bit more tautological - instead of the assertion coming from the tests, the assertion comes from a mix of the source code and the tests, which means that the tests don't validate a user-facing behavior so much as validating that the code is interacting with its own internal constants. 

I especially feel that way about this suggestion because the constants you mentioned aren't present in the source file, because I didn't need them. I would be exporting an internal constant that the source file doesn't even need for the sake of the tests, which is the kind of refactor that I've usually considered an antipattern.

I'll think on it for a little longer, but that's my first impression.

Reply:
You don't necessarily have to have the heavy rail lines in a module attribute, though I think you did before. If you're getting them from somewhere else then you can get them there in the test.

Reply:
> ...though I think you did before. 

This is exactly why I don't want to export the module attributes. I did a pretty hefty refactor to `Groups.groups/2` that changed which module attributes existed, and that refactor would have been harder if I had been exporting them. They're implementation details, and they change during refactors.

> I see that that's possible, but what is the benefit?

Bump on this question, which kind of feels like it makes all the difference.

Reply:
The benefit is that the test is more communicative.